fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -920,13 +920,12 @@ private void clearLockIfLocked() {
     @Override
     public void close() {
         try {
-            clearLockIfLocked();
+            clearLockIfLocked(false);
```

Comment:
Don't we close the directory after failures? Will this have the right behavior?

Reply:
If failed while a lock exists, `clearLockIfLocked(false)` will fail with "Already Closed" error, but the second attempt with true should succeed. 
Note that `abortAndClose()` can be called multiple times.